api/endpoints/algorithm.py

```diff
@@ -209,17 +211,28 @@ def post(self):
             return response_body, 500
 
         try:
-            git.update_git_repo(repo, repo_name=settings.REPO_NAME,
-                                algorithm_name=hysds.get_algorithm_file_name(algorithm_name))
-            # log.debug("Updated Git Repo with hash {}".format(commit_hash))
+            commit_hash = git.update_git_repo(repo, repo_name=settings.REPO_NAME,
+                                              algorithm_name=hysds.get_algorithm_file_name(algorithm_name))
+            logging.info("Updated Git Repo with hash {}".format(commit_hash))
         except Exception as ex:
             tb = traceback.format_exc()
             response_body["code"] = 500
-            response_body["message"] = "Failed to register {}".format(algorithm_name)
+            response_body["message"] = "Failed to register {}.".format(algorithm_name)
             response_body["error"] = "{} Traceback: {}".format(ex.message, tb)
             return response_body, 500
 
-        algorithm_id = "{}:{}".format(algorithm_name, req_data.get("code_version"))
+        try:
+            # Check and return the pipeline info and status
+            if commit_hash is None:
+                raise Exception("Commit Hash can not be None.")
+            gitlab_response = git.get_git_pipeline_status(project_id=settings.REGISTER_JOB_REPO_ID,
```

Comment:
I noticed the `get_git_pipeline_status` method waits for a response from the build pipeline. Do you have any concerns this might lead to a long running operation? If so, we may want to consider splitting this into a separate endpoint that the client can query after a successful registration submission.

Reply:
One idea to break the CI status method out from registration would be to return the commit hash as part of the registration response, then create a new "registration status" endpoint that accepts a commit hash for querying the CI engine.

Reply:
I don't have that concern. As soon as the pipeline  job is triggered on Gitlab, even if it waiting for the runner's availability to start the build we get a Job ID. So there's no hold up. I return the status and link to the Job ID which is used to show progress on the front end. So far we haven't had any issues on ImgSPEC.